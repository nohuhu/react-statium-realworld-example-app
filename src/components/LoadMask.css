.load-mask {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-color: #fff;
  opacity: 0.7;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: stretch;
}

.load-mask.hidden {
  display: none;
}

.load-mask .load-mask-inner {
  position: relative;
  text-align: center;
}

.load-mask .load-mask-inner .load-spinner {
  animation: spin 1s infinite linear;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(359deg);
  }
}
