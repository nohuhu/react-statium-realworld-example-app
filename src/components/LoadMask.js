import React from 'react';

import './LoadMask.css';

const LoadMask = ({ loading, size = '50px' }) => (
  <div className={`load-mask ${!loading ? 'hidden' : ''}`}>
    <div className="load-mask-inner">
      <i className="ion ion-load-c load-spinner"
        style={{ fontSize: size }} />
    </div>
  </div>
);

export default LoadMask;
