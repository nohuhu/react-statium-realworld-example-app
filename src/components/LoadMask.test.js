import React from 'react';
import { shallow } from 'enzyme';

import LoadMask from './LoadMask.jsx';

describe("LoadMask", () => {
  it("should render hidden by default", () => {
    const tree = shallow(<LoadMask />);

    expect(tree).toMatchSnapshot();
    expect(tree.find('div.load-mask').hasClass('hidden')).toBe(true);
  });

  it("should render visible when loading prop is true", () => {
    const tree = shallow(<LoadMask loading={true} />);

    expect(tree).toMatchSnapshot();
    expect(tree.find('div.load-mask').hasClass('hidden')).toBe(false);
  });
});
