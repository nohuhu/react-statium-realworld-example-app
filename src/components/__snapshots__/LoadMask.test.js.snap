// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMask should render hidden by default 1`] = `
<div
  className="load-mask hidden"
>
  <div
    className="load-mask-inner"
  >
    <i
      className="ion ion-load-c load-spinner"
      style={
        Object {
          "fontSize": "50px",
        }
      }
    />
  </div>
</div>
`;

exports[`LoadMask should render visible when loading prop is true 1`] = `
<div
  className="load-mask "
>
  <div
    className="load-mask-inner"
  >
    <i
      className="ion ion-load-c load-spinner"
      style={
        Object {
          "fontSize": "50px",
        }
      }
    />
  </div>
</div>
`;
